engineering/documents/CONTRIBUTING.md

```diff
@@ -0,0 +1,53 @@
+# Contributing Guidelines
+
+source{d} go-git project is [Apache 2.0 licensed](LICENSE) and accept
+contributions via GitHub pull requests.  This document outlines some of the
+conventions on development workflow, commit message formatting, contact points
+and other resources to make it easier to get your contribution accepted.
+
+## Certificate of Origin
+
+By contributing to this project you agree to the Developer Certificate of
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. See the [DCO](DCO) file for details.
+
+## Support Channels
+
+Whether you are a user or contributor, official support channels include:
+
+- GitHub [issues](https://github.com/src-d/go-git/issues/new)
+- Slack: #go-git room in the [source{d} Slack](http://sourced.tech/)
+
+Before opening a new issue or submitting a new pull request, it's helpful to
+search the project - it's likely that another user has already reported the
+issue you're facing, or it's a known issue that we're already aware of.
+
+
+## How to Contribute
+
+PRs are the main and exclusive way to contribute to the official go-git project.
+In order for a PR to be accepted it needs to pass a list of requirements:
+
+- You should be able to run the same query using `git`. We don't accept features that are not implemented in the official git implementation.
+- The expected behavior must match the official git implementation.
+- The actual behavior must be correctly explained with natural language and providing a minimum working example in Go that reproduces it.
+- In case a PR is submitted, it has to be written in idiomatic Go, formatted properly using gofmt, that compiles without warning and passes all the current tests.
```

Comment:
All PRs must be written in idiomatic Go, formatted according to gofmt, and without any warnings from `go lint` nor `go vet`.
They should in general include tests, and those shall pass.